tests/test_preview.py

```diff
@@ -0,0 +1,271 @@
+        osp.join("generator_files", "multi_tags_template.sh")
+    )
+    hello_world_model.attach_generator_files(to_configure=[config])
+    exp.generate(hello_world_model, overwrite=True)
+
+    preview_manifest = Manifest(hello_world_model, spam_eggs_model)
+
+    # Execute preview method
+    rendered_preview = previewrenderer.render(exp, preview_manifest)
+
+    # Evaluate output for hello_world_model
+    assert "Model name" in rendered_preview
+    assert "Executable" in rendered_preview
+    assert "Executable Arguments" in rendered_preview
+    assert "Batch Launch" in rendered_preview
+    assert "Model parameters" in rendered_preview
+    assert "Tagged Files for model configuration" in rendered_preview
+
+    assert "echo-hello" in rendered_preview
+    assert "/usr/bin/bash" in rendered_preview
+    assert "multi_tags_template.sh" in rendered_preview
+    assert "False" in rendered_preview
+    assert "port" in rendered_preview
+    assert "password" in rendered_preview
+    assert "6379" in rendered_preview
+    assert "unbreakable_password" in rendered_preview
+    assert "generator_files/multi_tags_template.sh" in rendered_preview
+
+    assert "echo-hello" == hello_world_model.name
+    assert "/usr/bin/bash" == hello_world_model.run_settings.exe[0]
+    assert "multi_tags_template.sh" == hello_world_model.run_settings.exe_args[0]
+    assert None == hello_world_model.batch_settings
+    assert "port" in list(hello_world_model.params.items())[0]
+    assert 6379 in list(hello_world_model.params.items())[0]
+    assert "password" in list(hello_world_model.params.items())[1]
+    assert "unbreakable_password" in list(hello_world_model.params.items())[1]
+    assert "generator_files/multi_tags_template.sh" in hello_world_model.files.tagged[0]
+
+    # Evaluate output for spam eggs model
+    assert "echo-spam" in rendered_preview
+    assert "/usr/bin/echo" in rendered_preview
+    assert "spam" in rendered_preview
+    assert "eggs" in rendered_preview
+    assert "echo-spam" == spam_eggs_model.name
+    assert "/usr/bin/echo" == spam_eggs_model.run_settings.exe[0]
+    assert "spam" == spam_eggs_model.run_settings.exe_args[0]
+    assert "eggs" == spam_eggs_model.run_settings.exe_args[1]
+
+
+def test_model_key_prefixing(fileutils, test_dir, wlmutils):
+    """
+    Test preview for enabling key prefixing for a Model
+    """
+    # Prepare entities
+    exp = Experiment("key_prefix_test", launcher="local")
+    db = exp.create_database(port=6780, interface="lo")
+    exp.generate(db, overwrite=True)
+    rs1 = exp.create_run_settings("echo", ["hello", "world"])
+    model = exp.create_model("model_test", run_settings=rs1)
+    # enable key prefixing on ensemble
+    model.enable_key_prefixing()
+    exp.generate(model, overwrite=True)
+
+    # Run preview render
+    preview_manifest = Manifest(db, model)
+    output = previewrenderer.render(exp, preview_manifest)
+
+    assert "Client Configuration" in output
```

Comment:
consider splitting tests per above logic:

- this named test would only look for key prefixing outputs
- I think ONLY line `253` w/ `... (key prefixing)` looks at prefix related output
  - is there anything else it can lok for?
  - should we check that the _prefix_ is in the output?

Reply:
separated tests